lua/sim/weapon.lua

```diff
@@ -506,15 +521,16 @@ Weapon = ClassWeapon(WeaponMethods, DebugWeaponComponent) {
 
         damageTable.__index = damageTable
 
+        self.damageTableCacheValid = true
+
         return damageTable
     end,
 
-    damageTableCache = false,
     ---@param self Weapon
     ---@return WeaponDamageTable
     GetDamageTable = function(self)
-        if not self.damageTableCache then
-            self.damageTableCache = self:GetDamageTableInternal()
+        if not self.damageTableCacheValid then
```

Comment:
Second check is for backwards compatibility.
First check would be for a new way that doesn't reset the entire damage table from the blueprint.
```suggestion
        if not self.damageTableCacheValid or not self.damageTableCache then
```